code/datums/components/obj_projectile_damage.dm

```diff
@@ -0,0 +1,32 @@
+/datum/component/obj_projectile_damage
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+
+/datum/component/obj_projectile_damage/Initialize()
+	if(!istype(parent, /obj))
+		return COMPONENT_INCOMPATIBLE
+	RegisterSignal(parent, COMSIG_ATOM_HITBY_PROJ, .proc/projectile_collide)
+	RegisterSignal(parent, COMSIG_ATOM_EXAMINE, .proc/examine)
+
+/datum/component/obj_projectile_damage/proc/projectile_collide(owner, var/obj/projectile/P)
+	var/obj/O = parent
+	if(P.proj_data.damage_type & (D_KINETIC | D_ENERGY | D_SLASHING))
+		if (O && (O._health/O._max_health) <= 0.5 && prob(abs((O._health/O._max_health) - 1) * 60))
+			var/obj/decal/cleanable/machine_debris/gib = null
+			gib = make_cleanable(/obj/decal/cleanable/machine_debris, O.loc)
+			gib.streak_cleanable()
+			elecflash(O, power = 2)
+		O.changeHealth(-round(((P.power/2)*P.proj_data.ks_ratio), 1.0))
```

Comment:
How many shots are we looking at to destroy something, roughly? As far as I can tell almost everything has `_max_health` of 100.

Reply:
Around 1/3 of the light machine-gun"s ammo

Reply:
Probably worth also using the Predator revolver as your measuring stick, I think it's probably the kinetic weapon used most often.

Reply:
4 shots with a predator revolver
24 shots with the nukie LMG
3 shots with the spes